core/network/src/main/java/com/example/network/interceptor/RateLimitInterceptor.kt

```diff
@@ -14,13 +20,19 @@ class RateLimitInterceptor : Interceptor {
             requestQueue.removeIf { it < now - TimeUnit.MINUTES.toMillis(1) }
 
             if (requestQueue.size >= maxRequestsPerMinute) {
```

Comment:
```suggestion
    rateLimitState.setRateLimitTriggered(requestQueue.size >= maxRequestsPerMinute)
      
    veya
    
    val isRateLimitTriggered = requestQueue.size >= maxRequestsPerMinute
    rateLimitState.setRateLimitTriggered(isRateLimitTriggered)
```
bu kullanımları buradaki if-else yerine tercih ederdim